standalone/test/unit_tests/data_conversion_layer/unittest_laserscan_conversions.cpp

```diff
@@ -201,37 +204,49 @@ TEST(LaserScanConversionsTest, shouldThrowProtocolErrorOnMissingFrames)
 
 TEST(LaserScanConversionsTest, shouldThrowProtocolErrorOnMismatchingThetaAngles)
 {
-  auto stamped_msgs = createStampedMsgs(6);
-  stamped_msgs[1] =
-      createStampedMsg(stamped_msgs[1].stamp_, stamped_msgs[1].msg_.fromTheta() + util::TenthOfDegree(10));
+  auto stamped_msgs = createValidStampedMsgs(6);
+  CHANGE_MSG_PROPERTY(stamped_msgs[1].msg_, fromTheta, util::TenthOfDegree(10));
   ASSERT_THROW(data_conversion_layer::LaserScanConverter::toLaserScan(stamped_msgs),
                data_conversion_layer::ScannerProtocolViolationError);
 }
 
 TEST(LaserScanConversionsTest, shouldThrowProtocolErrorOnMismatchingScanCounters)
 {
-  auto stamped_msgs = createStampedMsgs(6);
-  stamped_msgs[1] = createStampedMsg(stamped_msgs[1].stamp_,
-                                     stamped_msgs[1].msg_.fromTheta(),
-                                     stamped_msgs[1].msg_.resolution(),
-                                     stamped_msgs[1].msg_.scanCounter() + 1);
+  auto stamped_msgs = createValidStampedMsgs(6);
+  CHANGE_MSG_PROPERTY(stamped_msgs[1].msg_, scanCounter, 1);
   ASSERT_THROW(data_conversion_layer::LaserScanConverter::toLaserScan(stamped_msgs),
                data_conversion_layer::ScannerProtocolViolationError);
 }
 
 TEST(LaserScanConversionsTest, laserScanShouldContainAllScanInformationWhenBuildWithMultipleFrames)
 {
-  auto stamped_msgs = createStampedMsgs(6);
+  auto stamped_msgs = createValidStampedMsgs(6);
   std::unique_ptr<LaserScan> scan_ptr;
   ASSERT_NO_THROW(
       scan_ptr.reset(new LaserScan{ data_conversion_layer::LaserScanConverter::toLaserScan(stamped_msgs) }););
-  ASSERT_EQ(scan_ptr->getMeasurements().size(), stamped_msgs.size() * stamped_msgs[0].msg_.measurements().size());
-  ASSERT_EQ(scan_ptr->getIntensities().size(), stamped_msgs.size() * stamped_msgs[0].msg_.intensities().size());
+
+  auto measurements_it = scan_ptr->getMeasurements().begin();
+  auto intensities_it = scan_ptr->getIntensities().begin();
+  for (const auto& stamped_msg : stamped_msgs)
+  {
+    ASSERT_LE(measurements_it + stamped_msg.msg_.measurements().size(), scan_ptr->getMeasurements().end());
+    ASSERT_LE(intensities_it + stamped_msg.msg_.intensities().size(), scan_ptr->getIntensities().end());
+    EXPECT_THAT(
+        measurements_it,
+        UnsafePointwiseDoubleItEq(stamped_msg.msg_.measurements().begin(), stamped_msg.msg_.measurements().end()));
+    EXPECT_THAT(
+        intensities_it,
+        UnsafePointwiseDoubleItEq(stamped_msg.msg_.intensities().begin(), stamped_msg.msg_.intensities().end()));
+    measurements_it += stamped_msg.msg_.measurements().size();
+    intensities_it += stamped_msg.msg_.intensities().size();
+  }
+  EXPECT_EQ(measurements_it, scan_ptr->getMeasurements().end());
+  EXPECT_EQ(intensities_it, scan_ptr->getIntensities().end());
```

Comment:
What this test does is something like:

auto merged_measurements = MERGE_PROP_LISTS(measurements, stamped_msgs)
EXPECT_THAT(scan_ptr->measurements(), PointwiseDoubleEq(merged_measurements))

and the same for intensities? In tha case I would make 2 tests that do that in like 2 lines.

Reply:
I misunderstood this test, it checked that all scan information is contained by checking the vector sizes. I think this is fine.